worlds/yachtdice/__init__.py

```diff
@@ -0,0 +1,459 @@
+            ["Category Choice", "Category Double Threes and Fours"],
+            ["Category Inverse Choice", "Category Quadruple Ones and Twos"],
+            ["Category Ones", "Category Distincts"],
+            ["Category Twos", "Category Two times Ones"],
+            ["Category Threes", "Category Half of Sixes"],
+            ["Category Fours", "Category Twos and Threes"],
+            ["Category Fives", "Category Sum of Odds"],
+            ["Category Sixes", "Category Sum of Evens"],
+            ["Category Pair", "Category Micro Straight"],
+            ["Category Three of a Kind", "Category Three Odds"],
+            ["Category Four of a Kind", "Category 1-2-1 Consecutive"],
+            ["Category Tiny Straight", "Category Three Distinct Dice"],
+            ["Category Small Straight", "Category Two Pair"],
+            ["Category Large Straight", "Category 2-1-2 Consecutive"],
+            ["Category Full House", "Category Five Distinct Dice"],
+            ["Category Yacht", "Category 4&5 Full House"],
+        ]
+
+        # categories used in this game.
+        possible_categories = []
+
+        for index, cats in enumerate(all_categories):
+            possible_categories += [cats[categorylist[index]]]
+
+            # Add Choice and Inverse choice (or their alts) to the precollected list.
+            if index == 0 or index == 1:
+                self.precollected += [cats[categorylist[index]]]
+            else:
+                self.itempool += [cats[categorylist[index]]]
+
+        # Also start with one Roll and one Dice
+        self.precollected += ["Roll"]
+        self.precollected += ["Dice"]
+
+        # if one fragment per dice, just add "Dice" objects
+        if frags_per_dice == 1:
+            self.itempool += ["Dice"] * (num_of_dice - 1)  # minus one because one is in start inventory
+        else:
+            self.itempool += ["Dice"]  # always add a full dice to make generation easier (will be early)
+            self.itempool += ["Dice Fragment"] * (frags_per_dice * (num_of_dice - 2))
+
+        # if one fragment per roll, just add "Roll" objects
+        if frags_per_roll == 1:
+            self.itempool += ["Roll"] * (num_of_rolls - 1)  # minus one because one is in start inventory
+        else:
+            self.itempool += ["Roll"]  # always add a full roll to make generation easier (will be early)
+            self.itempool += ["Roll Fragment"] * (frags_per_roll * (num_of_rolls - 2))
+
+        already_items = len(self.itempool) + self.extra_plando_items
+
+        # Yacht Dice needs extra filler items so it doesn't get stuck in generation.
+        # For now, we calculate the number of extra items we'll need later.
+        if self.options.minimize_extra_items.value:
+            extra_percentage = max(0.1, 0.8 - self.multiworld.players / 10)
+        else:
+            extra_percentage = 0.7
+        extra_locations_needed = max(10, math.ceil(already_items * extra_percentage))
```

Comment:
I made this an option because reducing filler items might break if you join Yacht Dice with 6 Cliques. But any game with more than 2 locations should be fine.